docs/source/en/basics/router.md

```diff
@@ -0,0 +1,328 @@
+
+> **Note**: the verification is performed because the framework builds in a security plugin [egg-security](https://github.com/eggjs/egg-security) that provides some default security practices and this plugin is enabled by default. In case you want to disable some security protections, just configure the enable attribute of if to be false.
+
+> "Unless you know the consequence consciously, it's recommended not to disable functions provided by the security plugin"
+
+> Here we do the config temporarily in `config/config.default.js` for an example
+```
+exports.security = {
+  csrf: false
+};
+```
+
+### Form Verification
+
+```js
+// app/router.js
+module.exports = app => {
+  app.post('/user', 'user');
+};
+
+// app/controller/user.js
+const createRule = {
+  username: {
+    type: 'email',
+  },
+  password: {
+    type: 'password',
+    compare: 're-password',
+  },
+};
+
+exports.create = function* (ctx) {
+  // throws exceptions if the verification fails
+  ctx.validate(createRule);
+  ctx.body = ctx.request.body;
+};
+
+// curl -X POST http://127.0.0.1:7001/user --data 'username=abc@abc.com&password=111111&re-password=111111'
+```
+
+### Redirection
+
+#### Internal Redirection
+
+```js
+// app/router.js
+module.exports = app => {
+  app.get('index', '/home/index', 'home.index');
+  app.redirect('/', '/home/index', 302);
+};
+
+// app/controller/home.js
+exports.index = function* (ctx) {
+  ctx.body = 'hello controller';
+};
+
+// curl -L http://localhost:7001
+```
+
+#### External Redirection
+
+```js
+// app/router.js
+module.exports = app => {
+  app.get('/search', 'search');
+};
+
+// app/controller/search.js
+module.exports = function* () {
+  const type = this.query.type;
+  const q = this.query.q || 'nodejs';
+
+  if (type === 'bing') {
+    this.redirect(`http://cn.bing.com/search?q=${q}`);
+  } else {
+    this.redirect(`https://www.google.co.kr/search?q=${q}`);
+  }
+};
+
+// curl http://localhost:7001/search?type=bing&q=node.js
+// curl http://localhost:7001/search?q=node.js
+```
+
+### Using Middleware
+
+A middleware can be used to change the request parameter of one kind of user requests to uppercase.
+Here we show how to use the middleware briefly, referring to [Middleware](./middleware.md) for detail.
+
+```js
+// app/controller/search.js
+module.exports = function* (ctx) {
+  ctx.body = `search: ${this.query.name}`;
+};
+
+// app/middleware/uppercase.js
+module.exports = () => {
+  return function* (next) {
+    this.query.name = this.query.name && this.query.name.toUpperCase();
+    yield next;
+  };
+};
+
+// app/router.js
+module.exports = app => {
+  app.get('s', '/search', app.middlewares.uppercase(), 'search')
+};
+
+// curl http://localhost:7001/search2?name=egg
+```
+
+### Too Many Routing Maps?
+
+As described above, we do not recommend you scatter routing logics all around, or it leads to a difficult trouble shooting.
```

Comment:
'we do not recommend ...'
suggestion: 
'we do not recommend that ..., or it will bring trouble in trouble shooting'